### PR TITLE
Add compiler plugin continuation to project settings

### PR DESCRIPTION
**Pull request by kasper-f**
I have today made a fork and implemented a solution for adding continuation compiler plugin to idea config. See bug https://github.com/mpeltonen/sbt-idea/issues/102

Only problem is that I am not sure the path is calculated the correct way or if I somehow could get access to the "addCompilerPlugin" lib defined in the build. The above solution works for me on linux.

Do you see any problems including this in please let me know and I will try solving it

### CODE REVIEW

**Comment by mpeltonen** [comment]
Thanks for the patch. I think it would be better to be more general here however, and not restrict this to continuations plugin.

It looks like you can see all the plugins in scalacOptions if you look it in Compile scope:

``` code
> show compile:scalac-options
[info] List(-P:continuations:enable, -Xplugin:/Users/mpe/.ivy2/cache/org.scala-lang.plugins/continuations/jars/continuations-2.10.0.jar, -Xplugin:/Users/mpe/.ivy2/cache/org.scala-tools.sxr/sxr_2.9.0/jars/sxr_2.9.0-0.2.7.jar)
```

project.scalacOptions could be changed to use Compile scope like this in SbtIdeaPlugin:

``` code
-    val scalacOptions: Seq[String] = settings.optionalTask(Keys.scalacOptions).getOrElse(Seq())
+    val scalacOptions: Seq[String] = settings.optionalTask(Keys.scalacOptions in Compile).getOrElse(Seq())
```

This way you could loop all scalacOptions that start with -Xplugin, parse the path from there and add all plugins with correct path to xml.

This also means you have to filter out -Xplugin -options when setting scalacOptions to Idea config (IdeaModuleDescriptor line 157).

Hope this helps.

**Comment by kasper-f** [comment]
Thanks for the comments, I am pretty new SBT so still learning;-) 

I have tried to implement it as you suggested but have also made some other changes to the scala facet generation:

I have separated the mapping from scalacOption to xml tag into a single map and then uses the map to map the entire scalacOption seq with. 

I filter out options defined in idea from the idea compilerOption option (see note in the diff)

Hope it fits better now

File: src/main/scala/org/sbtidea/IdeaModuleDescriptor.scala [hunk 1]
```diff
@@ -149,12 +158,25 @@ class IdeaModuleDescriptor(val imlDir: File, projectRoot: File, val project: Sub
...
+                {
+                project.scalacOptions.filter(x=> x.startsWith("-Xplugin:")).map(x=>x.substring(9)).
+                  map(x=> <option value={x} />)
+                }
+              </array>
+            </option>
+          }
+        }
+        {
+        val options=nonXplugin.filter(x=> !optionMap.contains(x)).mkString(" ")
```
Comment (kasper-f) [inline src/main/scala/org/sbtidea/IdeaModuleDescriptor.scala:176]: uses the seq where xPlugin is filtered out and filteres out all options defined by other idea options defined in the optionMap

File: src/main/scala/org/sbtidea/IdeaModuleDescriptor.scala [hunk 2]
```diff
@@ -149,12 +158,25 @@ class IdeaModuleDescriptor(val imlDir: File, projectRoot: File, val project: Sub
...
+          nonXplugin.filter(x=> optionMap.contains(x)).map(x=> optionMap(x))
         }
         {
-          if (project.scalacOptions.contains("-unchecked")) <option name="uncheckedWarnings" value="true" />
+          val xplugin = project.scalacOptions.filter(x=> x.startsWith("-Xplugin:")).map(x=>x.substring(9))
+          if (!xplugin.isEmpty){
+            <option name="pluginPaths">
+              <array>
+                {
+                project.scalacOptions.filter(x=> x.startsWith("-Xplugin:")).map(x=>x.substring(9)).
```
Comment (jozic) [inline src/main/scala/org/sbtidea/IdeaModuleDescriptor.scala]: can "xplugin" be used here instead?

**Comment by mpeltonen** [comment]
I cherry-picked your changes, added a test and made some minor code cleanups. Test if 1.4.0-SNAPSHOT works for you or if you find some issues!

Thanks for the patch!

**Pull request closed without merging**